tests/test_packages/test_protocols/test_ml_message.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests of the ml_messages module."""
+import logging
+from unittest import mock
+
+import pytest
+
+from aea.protocols.oef.models import DataModel, Attribute, Constraint, Query, ConstraintType, Description
+from packages.protocols.ml_trade.message import MLTradeMessage
+from packages.protocols.ml_trade.serialization import MLTradeSerializer
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def test_perfomrative_str():
+    """Test the str value of each performative."""
+    assert str(MLTradeMessage.Performative.CFT) == 'cft'
+    assert str(MLTradeMessage.Performative.TERMS) == 'terms'
+    assert str(MLTradeMessage.Performative.ACCEPT) == 'accept'
+    assert str(MLTradeMessage.Performative.DATA) == 'data'
+
+
+def test_ml_wrong_message_creation():
+    """Test the creation of a ml message."""
+    with pytest.raises(AssertionError):
+        MLTradeMessage(performative=MLTradeMessage.Performative.CFT, query="")
+
+
+def test_ml_message_creation():
+    """Test the creation of a ml message."""
+    dm = DataModel("ml_datamodel", [Attribute("dataset_id", str, True)])
+    query = Query([Constraint("dataset_id", ConstraintType("==", "fmnist"))], model=dm)
+    msg = MLTradeMessage(performative=MLTradeMessage.Performative.CFT, query=query)
+    msg_bytes = MLTradeSerializer().encode(msg)
+    reverted_msg = MLTradeSerializer().decode(msg_bytes)
```

Comment:
I think you mean `recovered` rather than `reverted`?

Reply:
It would be better to split into individual tests next time, for now I'd leave it.